src/server/engine_shard_set.cc

```diff
@@ -868,6 +870,8 @@ ShardId Shard(string_view v, ShardId shard_num) {
       has_hashtags = true;
       v = v_hash_tag;
     }
+  } else if (absl::GetFlag(FLAGS_enable_round_robin_sharding)) {
```

Comment:
Originally, when I implemented this, Roman said that it should only be implemented for hash tags because we don't know of a scenario that it will be used without it.
Now we know of such a scenario :)
As such, I'd propose to remove the new flag. It's enough that we require users to specify `--shard_round_robin_prefix`, I don't think they should also enable it via another flag.
Please modify the flag desc and say that if key has a hashtag, the prefix is matched against the hash-tag part.

Reply:
+1 to @chakaz suggestion.